### PR TITLE
dockerfile, first db creation, search example

### PR DESCRIPTION
**Pull request by henderkes**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# escape=`
+
+RUN pip install pymongo
+RUN pip install Flask
+
+FROM microsoft/windowsservercore as msi
```
Comment (Drakulix) [inline Dockerfile]: Why are we using a windows image as a base? Does this even work outside of windows?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+RUN pip install Flask
+
+FROM microsoft/windowsservercore as msi
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+ENV MONGO_VERSION 3.4.4
+ENV MONGO_DOWNLOAD_URL http://downloads.mongodb.org/win32/mongodb-win32-x86_64-2008plus-ssl-${MONGO_VERSION}-signed.msi
+ENV MONGO_DOWNLOAD_SHA256 73d1121a1fe58fe29c4fb7b6f4b79a4d33afa666f357960419ef83635396f977
+
```
Comment (Drakulix) [inline Dockerfile]: There is already a mongo docker image, no need to replicate it https://hub.docker.com/_/mongo/

**Review by Drakulix — CHANGES_REQUESTED** [review]
We probably only need the `Dockerfile` for our own python webserver (that container should be based on the official [`python` docker image](https://hub.docker.com/_/python/) and the rest of the services in a [`docker-compose.yaml`](https://docs.docker.com/compose/).

The `first.py` is a good start and helpful, but please use a proper name (maybe main.py) and create a folder for it. (See #6)

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# escape=`
+
+RUN pip install pymongo
+RUN pip install Flask
+
```
Comment (Drakulix) [inline Dockerfile]:
```
Step 1/22 : RUN pip install pymongo
Please provide a source image with `from` prior to run
```

Cannot even test it..

**Comment by henderkes** [comment]
updated

File: LICENSE [hunk 4]
```diff
@@ -1,21 +0,0 @@
-MIT License
```
Comment (Drakulix) [inline LICENSE]: ähm, the license is not an "unnecessary file".
Your first commit does not seem right.
Please try to do the following steps:

```
git checkout master
git pull
git checkout DubbleClick-patch-1
git rebase -i master
```

And remove your [last commit](6cf8cd0bd00f9ca61d1350be19386f2fb7e2bb1b) (skip) from the rebase.
Also please skip all commits, that are already part of master (all travis related commits) and then fix any remaining conflicts, if any (should all be fine).

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+version: '3'
+
+services:
+    mongodb-service:
+        build: ./mongodb
```
Comment (Drakulix) [inline docker-compose.yml]: This is unnecessary. You can just do `image: mongo`, that's it.

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+        ports:
+            - 27017:27017
```
Comment (Drakulix) [inline docker-compose.yml]: Unnecessary, please see the next comment.

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    python-file:
+        build: ./pythonfile
+        ports:
+            - 8000:27017
```
Comment (Drakulix) [inline docker-compose.yml]: Please just do:

```
links:
    - mongodb-service
```
to automatically route all mongodb ports to only the python container and just open the port of the outgoing webserver port of flask.

File: docker-compose.yml [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+        depends_on:
+            - mongodb-service
```
Comment (Drakulix) [inline docker-compose.yml]: Please add a trailing new line

File: mongodb/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,25 @@
+# Pull base image.
```
Comment (Drakulix) [inline mongodb/Dockerfile]: File is unnecessary (see comment regarding `image:` key in docker-compose.yml)

File: pythonfile/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,2 @@
+FROM python:3.6.1
```
Comment (Drakulix) [inline pythonfile/Dockerfile]: Please use `3` instead of `3.6.1`.

File: pythonfile/requirements.txt [hunk 11]
```diff
@@ -0,0 +1,2 @@
+flask
+pymongo
```
Comment (Drakulix) [inline pythonfile/requirements.txt:2]: :+1:

File: pythonfile/Dockerfile [hunk 12]
```diff
@@ -0,0 +1,2 @@
...
+CMD ["python", "first.py"]
```
Comment (Drakulix) [inline pythonfile/Dockerfile]: Example taken from the python docker image [hub website](https://hub.docker.com/_/python/):

```
FROM python:3

WORKDIR /usr/src/app

COPY requirements.txt ./
RUN pip install --no-cache-dir -r requirements.txt

COPY . .

CMD [ "python", "./your-daemon-or-script.py" ]
```

By coping you can get the source files into the image (also fixing your windows problem / see slack) and redistribute our whole app as a container.
Also running `pip` will install all dependencies.

**Review by Drakulix — CHANGES_REQUESTED** [review]
Thanks! Looks already very good. Just some nitpicks and smaller details.

**Review by Drakulix — DISMISSED** [review]
Great!

You git history still seems a little confusing, although all files are here.
(I will fix that myself and merge asap)

**Review by Drakulix — APPROVED** [review]
@DubbleClick I have rebased your work. Can you check, that everything is still there?

Afterwards it's ready for merge.

**Pull request merged**